ui/components/multichain/notification-detail-button/notification-detail-button.tsx

```diff
@@ -46,17 +56,21 @@ export const NotificationDetailButton = ({
         ...('chain_id' in notification && {
           chain_id: notification.chain_id,
         }),
-        clicked_item: 'block_explorer',
+        clicked_item: getClickedItem(),
       },
     });
+
+    if (notification.type === TRIGGER_TYPES.SNAP && isMetaMaskUrl) {
+      navigate(href);
+    }
   };
 
   return (
     <Button
       key={id}
-      href={href}
+      href={!isMetaMaskUrl && href ? href : undefined}
       variant={variant}
-      externalLink={isExternal}
+      externalLink={!isMetaMaskUrl}
```

Comment:
Seems like this would break existing uses of this component with `isExternal=false`?

Reply:
Yeah, fixed.